src/Core/System/SalesChannel/Api/ResponseFields.php

```diff
@@ -10,41 +10,81 @@ class ResponseFields
 {
     /**
      * @param array<string, list<string>>|null $includes
+     * @param array<string, list<string>>|null $excludes
      */
-    public function __construct(protected ?array $includes = null)
-    {
+    public function __construct(
+        protected ?array $includes = null,
+        protected ?array $excludes = null,
+    ) {
+        $this->validateFields();
     }
 
     public function isAllowed(string $type, string $property): bool
     {
-        if (!isset($this->includes[$type])) {
-            return true;
+        if (isset($this->excludes[$type]) && \in_array($property, $this->excludes[$type], true)) {
+            return false;
         }
 
-        if (!\is_array($this->includes[$type])) {
-            throw SalesChannelException::invalidType(
-                \sprintf(
-                    'The includes for type "%s" must be of the type array, %s given',
-                    $type,
-                    \gettype($this->includes[$type])
-                )
-            );
+        if (isset($this->includes[$type])) {
+            return \in_array($property, $this->includes[$type], true);
         }
 
-        return \in_array($property, $this->includes[$type], true);
+        return true;
     }
 
     public function hasNested(string $alias, string $prefix): bool
     {
-        $fields = $this->includes[$alias] ?? [];
-
         $prefix .= '.';
-        foreach ($fields as $property) {
+
+        $excludeFields = $this->excludes[$alias] ?? [];
+        foreach ($excludeFields as $property) {
+            if (str_starts_with((string) $property, $prefix)) {
+                return false;
+            }
+        }
+
+        $includeFields = $this->includes[$alias] ?? [];
+        foreach ($includeFields as $property) {
             if (str_starts_with((string) $property, $prefix)) {
                 return true;
             }
         }
 
         return false;
     }
+
+    protected function validateFields(): void
+    {
+        if (isset($this->includes)) {
+            foreach ($this->includes as $type => $fields) {
+                if (\is_array($fields)) {
+                    continue;
+                }
+
+                throw SalesChannelException::invalidType(
+                    \sprintf(
+                        'The includes for type "%s" must be of the type array, %s given',
+                        $type,
+                        \gettype($fields)
+                    )
+                );
+            }
+        }
+
+        if (isset($this->excludes)) {
```

Comment:
```suggestion
        if (is_array($this->excludes)) {
```

I like more to check for what I need exactly